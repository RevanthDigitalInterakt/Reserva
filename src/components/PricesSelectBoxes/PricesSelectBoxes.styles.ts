import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  textWrapper: {
    fontWeight: '400',
    marginHorizontal: 16,
    fontSize: 18,
  },
});
