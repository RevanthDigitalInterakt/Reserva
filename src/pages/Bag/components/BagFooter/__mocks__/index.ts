import { profileQuery } from '../../../../../graphql/profile/profileQuery';

export const bagInfos = {
  totalBagItemsPrice: 100,
  totalBagItems: 20,
  totalBagDiscountPrice: 40,
  totalBagDeliveryPrice: 60,
};

export const currentOrderForm: any = {
  items: [
    {
      name: 'Produto A',
      skuName: 'test',
      price: 50.0,
      quantity: 2,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
    {
      name: 'Produto B',
      skuName: 'test',
      price: 20.0,
      quantity: 1,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
  ],
  orderFormId: '1',
};

export const installmentInfo = {
  installmentPrice: 110,
  installmentsNumber: 40,
  totalPrice: 1000,
};

export const apolloMocks = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          addresses: [[{}]],
          customFields: [[{}], [{}], [{}]],
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: 'Tester',
          lastName: 'Silva',
          fullName: 'teste da Silva',
          email: 'nogueirahy@gmail.com',
          document: '11111111111',
          birthDate: '1980-01-01T00:00:00.000Z',
          gender: 'male',
          homePhone: '+5511991111111',
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];

export const apolloMocksWithoutDataUser = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: null,
          lastName: null,
          fullName: null,
          email: 'nogueirahy@gmail.com',
          document: null,
          birthDate: null,
          gender: 'male',
          homePhone: null,
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];
