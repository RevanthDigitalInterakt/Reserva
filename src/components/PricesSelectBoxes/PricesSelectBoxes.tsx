import React, {
  useLayoutEffect, useState, useCallback,
} from 'react';
import { View } from 'react-native';
import { Typography } from '@usereservaapp/reserva-ui';

import testProps from '../../utils/testProps';
import { useCart } from '../../context/CartContext';
import { useAuthStore } from '../../zustand/useAuth/useAuthStore';
import type { ProductSizeOutput } from '../../base/graphql/generated';
import { usePrimeStore } from '../../zustand/usePrimeStore/usePrimeStore';

import { ModalSignIn } from '../ModalSignIn';
import { ModalBag } from '../ModalBag/ModalBag';
import { ModalWelcomePrime } from '../ModalWelcomePrime';
import { SelectBoxPrime } from '../SelectBoxPrime/SelectBoxPrime';
import { SelectBoxNormal } from '../SelectBoxNormal/SelectBoxNormal';

import { styles } from './PricesSelectBoxes.styles';
import { usePrimeInfo } from '../../hooks/usePrimeInfo';

interface IPropsPriceSelectBoxes {
  selectedSize: ProductSizeOutput | null;
}

function PricesSelectBoxes({ selectedSize }: IPropsPriceSelectBoxes) {
  const [isNormalPriceSelected, setIsNormalPriceSelected] = useState(false);
  const [isPrimePriceSelected, setIsPrimePriceSelected] = useState(false);
  const [isModalSignInVisible, setIsModalSignInVisible] = useState(false);
  const { isPrime } = usePrimeInfo();

  const {
    animationBag,
    handleClickContinue,
    isVisibleModalWelcome,
    changeStateAnimationBag,
    changeStateIsVisibleModalWelcome,
  } = usePrimeStore([
    'animationBag',
    'handleClickContinue',
    'isVisibleModalWelcome',
    'changeStateAnimationBag',
    'changeStateIsVisibleModalWelcome',
  ]);

  const { orderForm } = useCart();
  const { profile } = useAuthStore(['profile']);

  const hasDiscount = selectedSize?.hasDiscount;

  const selectNormalPrice = useCallback(() => {
    setIsNormalPriceSelected(true);
    setIsPrimePriceSelected(false);
  }, []);

  const selectPrimePrice = useCallback(() => {
    setIsNormalPriceSelected(false);
    setIsPrimePriceSelected(true);
  }, []);

  const selectPriceBasedOnUser = useCallback(() => {
    if (!hasDiscount && (profile?.isPrime || isPrime)) {
      selectPrimePrice();
    } else {
      selectNormalPrice();
    }
  }, [hasDiscount, isPrime, profile?.isPrime]);

  useLayoutEffect(() => {
    selectPriceBasedOnUser();
  }, [
    hasDiscount,
    orderForm,
    isPrime,
    isModalSignInVisible,
    selectPriceBasedOnUser,
  ]);

  const handleSelectBoxesPress = useCallback(
    (option: string) => {
      if (option === 'priceNormal' && !isNormalPriceSelected) {
        selectNormalPrice();
      } else if (option === 'pricePrime' && !isPrimePriceSelected) {
        selectPrimePrice();

        if (!profile?.isPrime && !isPrime) {
          setIsModalSignInVisible(true);
        }
      }
    },
    [
      isPrime,
      isNormalPriceSelected,
      isPrimePriceSelected,
      profile?.isPrime,
    ],
  );

  const savedValueProduct = (): number => {
    const currentPriceProduct = selectedSize?.currentPrice || 0;
    const pricePrime = selectedSize?.prime?.price || 0;

    if (!currentPriceProduct && !pricePrime) return 0;

    return currentPriceProduct - pricePrime;
  };

  const handleOnModalHide = useCallback(() => {
    if (isPrime) {
      changeStateIsVisibleModalWelcome(true);
    }
  }, [changeStateIsVisibleModalWelcome, isPrime]);

  const handleOnModalHideSignIn = useCallback(() => {
    if (isPrime) {
      changeStateAnimationBag(true);
    }
  }, [changeStateAnimationBag, isPrime]);

  return (
    <View {...testProps('com.usereserva:id/prices_select_boxes')}>
      <ModalSignIn
        isVisible={isModalSignInVisible}
        onClose={() => setIsModalSignInVisible(false)}
        onModalHide={handleOnModalHideSignIn}
      />

      <ModalWelcomePrime isVisible={isVisibleModalWelcome} onClose={handleClickContinue} />

      <ModalBag
        isVisible={animationBag}
        onBackdropPress={() => changeStateAnimationBag(false)}
        onModalHide={handleOnModalHide}
      />

      <SelectBoxNormal
        installmentsNumber={selectedSize?.installment.number ?? 1}
        installmentsPrice={selectedSize?.installment.value ?? 0}
        isChecked={isNormalPriceSelected}
        onPress={handleSelectBoxesPress}
        price={
          hasDiscount ? selectedSize.currentPrice : selectedSize?.listPrice
        }
      />

      {selectedSize?.hasDiscount ? (
        <Typography
          fontFamily="reservaSansRegular"
          color="fullBlack"
          style={styles.textWrapper}
        >
          O desconto do
          <Typography
            fontFamily="reservaDisplayRegular"
            variant="descontoTag2"
            fontSize={20}
            color="vermelhoRSV"
          >
            {' '}
            Prime
            {' '}
          </Typography>
          não é cumulativo com produtos em liquidação.
        </Typography>
      ) : (
        <SelectBoxPrime
          installmentsNumber={selectedSize?.prime?.installment?.number ?? 1}
          installmentsPrice={selectedSize?.prime?.installment?.value ?? 0}
          isChecked={isPrimePriceSelected}
          onPress={handleSelectBoxesPress}
          savedValue={savedValueProduct()}
          price={selectedSize?.prime?.price}
        />
      )}
    </View>
  );
}

export default PricesSelectBoxes;
