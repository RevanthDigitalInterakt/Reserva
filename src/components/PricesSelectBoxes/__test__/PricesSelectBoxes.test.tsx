import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import {
  fireEvent, render, screen,
} from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import PricesSelectBoxes from '../PricesSelectBoxes';
import { ApolloMockLPPrime } from '../../../pages/PrimeLP/__mocks__/primeLPMocks';
import * as useLandingPagePrimeQuery from '../../../base/graphql/generated';
import { mockPrimeData } from '../../../../__mocks__/PrimeLP.mock';
import { theme } from '../../../base/usereservappLegacy/theme';

const selectedSizeMock = {
  __typename: 'ProductSizeOutput',
  availableQuantity: 140,
  currentPrice: 349,
  disabled: false,
  discountPercent: 0,
  ean: '0070649312',
  hasDiscount: false,
  installment: {
    __typename: 'ProductSizeInstallmentOutput',
    number: 5,
    value: 69.8,
  },
  itemId: '434252',
  listPrice: 349,
  seller: '1',
  size: 'M',
  prime: {
    installment: {
      number: 4,
      value: 69.8,
    },
    price: 279.2,
  },
};

const selectedSizeWithDiscountMock = {
  __typename: 'ProductSizeOutput',
  availableQuantity: 140,
  currentPrice: 349,
  disabled: false,
  discountPercent: 0,
  ean: '0070649312',
  hasDiscount: true,
  installment: {
    __typename: 'ProductSizeInstallmentOutput',
    number: 5,
    value: 69.8,
  },
  itemId: '434252',
  listPrice: 349,
  seller: '1',
  size: 'M',
  prime: {
    installment: {
      number: 4,
      value: 69.8,
    },
    price: 279.2,
  },
};

jest.mock('../../../zustand/useApolloFetchPolicyStore', () => ({
  useApolloFetchPolicyStore: () => ({
    getFetchPolicyPerKey: () => 'network-only',
  }),
}));

jest
  .spyOn(useLandingPagePrimeQuery, 'useLandingPagePrimeQuery')
  .mockReturnValue({
    data: {
      landingPagePrime: {
        ...mockPrimeData,
      },
    },
    loading: false,
  } as any);

function TestingComponent(selectedSize: any) {
  return (
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={ApolloMockLPPrime} addTypename={false}>
        <PricesSelectBoxes selectedSize={selectedSize} />
      </MockedProvider>
    </ThemeProvider>
  );
}

describe('PricesSelectBoxes', () => {
  beforeEach(() => {
    jest.useFakeTimers({ legacyFakeTimers: true });
  });

  it('should render properly', () => {
    render(TestingComponent(selectedSizeMock));

    const pricesSelectBoxes = screen.getByTestId(
      'com.usereserva:id/prices_select_boxes',
    );

    expect(pricesSelectBoxes).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    render(TestingComponent(selectedSizeMock));
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should show text that prime is not cumulative if has discount', () => {
    const { rerender } = render(TestingComponent(selectedSizeWithDiscountMock));

    const text = screen.getByText(
      'O desconto do Prime não é cumulativo com produtos em liquidação.',
    );

    expect(text).toBeOnTheScreen();
  });

  it('should change selection when a price option is pressed', () => {
    const { rerender } = render(TestingComponent(selectedSizeMock));

    const selectBoxNormal = screen.getByTestId('com.usereserva:id/select_box_price_normal');
    const selectBoxPrime = screen.getByTestId('com.usereserva:id/select_box_price_prime');

    expect(selectBoxNormal).toBeOnTheScreen();
    expect(selectBoxPrime).toBeOnTheScreen();

    fireEvent.press(selectBoxNormal);

    // rerender component
    rerender(TestingComponent(selectedSizeMock));

    const checkedNormal = screen.getByTestId('com.usereserva:id/select_box_price_normal_checked');

    // verifying SelectBoxNormal is checked
    expect(checkedNormal).toBeOnTheScreen();

    fireEvent.press(selectBoxPrime);

    rerender(TestingComponent(selectedSizeMock));

    const ModalSignIn = screen.getByTestId('com.usereserva:id/modal_sign_in');

    expect(ModalSignIn).toBeOnTheScreen();
  });
});
