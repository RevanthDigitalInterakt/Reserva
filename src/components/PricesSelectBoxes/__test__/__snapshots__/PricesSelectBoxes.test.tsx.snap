// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricesSelectBoxes should match to snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/prices_select_boxes"
  testID="com.usereserva:id/prices_select_boxes"
>
  <Modal
    accessibilityLabel="com.usereserva:id/modal_sign_in"
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    testID="com.usereserva:id/modal_sign_in"
    transparent={true}
    visible={false}
  />
  <Modal
    accessibilityLabel="com.usereserva:id/modal_welcome_prime"
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    testID="com.usereserva:id/modal_welcome_prime"
    transparent={true}
    visible={false}
  />
  <View
    style={{}}
  >
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      testID="com.usereserva:id/modal_bag"
      transparent={true}
      useNativeDriverForBackdrop={true}
      visible={false}
    />
  </View>
  <View
    accessibilityLabel="com.usereserva:id/select_box_price_normal"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginHorizontal": 16,
        "marginVertical": 12,
        "opacity": 1,
      }
    }
    testID="com.usereserva:id/select_box_price_normal"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#8D8D8D",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#000000",
            "borderRadius": 11,
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "marginRight": 0,
            "padding": 3,
            "width": 22,
          }
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/select_box_price_normal_checked"
          style={
            {
              "backgroundColor": "#000000",
              "borderRadius": 8,
              "height": 16,
              "width": 16,
            }
          }
          testID="com.usereserva:id/select_box_price_normal_checked"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "paddingVertical": 10,
          }
        }
      >
        <Text
          fontFamily="reservaSansRegular"
          style={
            [
              {
                "fontFamily": "ReservaSans-Regular",
              },
              {
                "color": "#8D8D8D",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          Pre√ßo Normal
        </Text>
        <View
          style={
            {
              "alignItems": "baseline",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 4,
            }
          }
        >
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#8D8D8D",
                  "fontSize": 18,
                  "fontWeight": "400",
                },
              ]
            }
          >
            5
            x
             
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#8D8D8D",
                  "fontSize": 24,
                  "fontWeight": "700",
                },
              ]
            }
          >
            R$
             
            69,
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#8D8D8D",
                  "fontWeight": "700",
                },
              ]
            }
          >
            80
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#8D8D8D",
            "height": 30,
            "marginHorizontal": 0,
            "marginTop": 30,
            "width": 1,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "baseline",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 50,
              "marginTop": 24,
            }
          }
        >
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#8D8D8D",
                  "fontSize": 24,
                  "fontWeight": "700",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            R$
             
            349,
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            fontWeight="bold"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                  "fontWeight": "700",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            00
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/select_box_price_prime"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginHorizontal": 16,
        "opacity": 1,
      }
    }
    testID="com.usereserva:id/select_box_price_prime"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fffdeb",
          "borderColor": "#c4a968",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#c4a968",
            "borderRadius": 11,
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "marginRight": 0,
            "padding": 3,
            "width": 22,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "marginLeft": 16,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          fontFamily="reservaSansRegular"
          style={
            [
              {
                "fontFamily": "ReservaSans-Regular",
              },
              {
                "color": "#9e7e2f",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          Clientes
          <Text
            fontSize={20}
            style={
              {
                "fontFamily": "ReservaSerif-Bold",
                "fontSize": 20,
              }
            }
            variant="descontoTag2"
          >
             
            Prime
          </Text>
        </Text>
        <View
          style={
            {
              "alignItems": "baseline",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 4,
            }
          }
        >
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#9e7e2f",
                  "fontSize": 18,
                  "fontWeight": "400",
                },
              ]
            }
          >
            4
            x
             
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#9e7e2f",
                  "fontSize": 24,
                  "fontWeight": "700",
                },
              ]
            }
          >
            R$
             
            69,
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            fontWeight="bold"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                  "fontWeight": "700",
                },
                {
                  "color": "#9e7e2f",
                },
              ]
            }
          >
            80
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#8D8D8D",
            "height": 30,
            "marginHorizontal": 15,
            "marginTop": 30,
            "width": 1,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "marginLeft": 6,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "height": 28,
              "justifyContent": "center",
              "paddingLeft": 10,
              "width": 160,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              {
                "commons": {
                  "GuideBermuda": [Circular],
                  "GuideCalca": [Circular],
                  "GuideCamiseta": [Circular],
                  "GuideHoodie": [Circular],
                  "GuideJaqueta": [Circular],
                  "GuideMangaCurta": [Circular],
                  "GuideMangaLonga": [Circular],
                  "GuidePolo": [Circular],
                  "GuideShort": [Circular],
                  "GuideSueter": [Circular],
                  "GuideSungaCueca": [Circular],
                  "QrcodeBackground": [Circular],
                  "abandonedCartBag": [Circular],
                  "addition": [Circular],
                  "americanExpress": [Circular],
                  "arrowInstruction": [Circular],
                  "arrowRight": [Circular],
                  "bagEmpty": [Circular],
                  "bannerCatalog": [Circular],
                  "bannerLogin": [Circular],
                  "bannerOffer": [Circular],
                  "black1p5p": [Circular],
                  "cardFront": [Circular],
                  "cardImage": [Circular],
                  "cardVerso": [Circular],
                  "chat": [Circular],
                  "checkedBox": [Circular],
                  "chevronLeftFill": [Circular],
                  "chevronRightFill": [Circular],
                  "close": [Circular],
                  "clothingCare1": [Circular],
                  "clothingCareBleach": [Circular],
                  "clothingCareDrying": [Circular],
                  "clothingCarePass": [Circular],
                  "clothingCareProfessionalWashing": [Circular],
                  "clothingCareWash": [Circular],
                  "config": [Circular],
                  "copy": [Circular],
                  "cupomModalBackground": [Circular],
                  "cutlery": [Circular],
                  "discountFlag": [Circular],
                  "dividerReserva": [Circular],
                  "document": [Circular],
                  "elo": [Circular],
                  "emptyProductCatalog": [Circular],
                  "expand": [Circular],
                  "facebook": [Circular],
                  "faq": [Circular],
                  "filledHeart": [Circular],
                  "foraDoAr": [Circular],
                  "green1p5p": [Circular],
                  "greenCheck": [Circular],
                  "headerLogin": [Circular],
                  "heart": [Circular],
                  "help": [Circular],
                  "homeMock1": [Circular],
                  "homeMock2": [Circular],
                  "homeMock3": [Circular],
                  "homeMock4": [Circular],
                  "imageCashback": [Circular],
                  "imageNotFound": [Circular],
                  "instagram": [Circular],
                  "ios14": [Circular],
                  "localReserva": [Circular],
                  "logo": [Circular],
                  "logoQRCode": [Circular],
                  "mastercard": [Circular],
                  "newsletter": [Circular],
                  "noStoresFound": [Circular],
                  "noWishList": [Circular],
                  "pinPlace": [Circular],
                  "pinYou": [Circular],
                  "profile": [Circular],
                  "purchaseConfirmation": [Circular],
                  "raceImageBackground": [Circular],
                  "reservaIcon": [Circular],
                  "roulet": [Circular],
                  "saleOff": [Circular],
                  "search": [Circular],
                  "searchNotFound": [Circular],
                  "selectRectangle": [Circular],
                  "share": [Circular],
                  "shipping1": [Circular],
                  "shipping2": [Circular],
                  "shirt1": [Circular],
                  "shirt2": [Circular],
                  "shirt3": [Circular],
                  "shirt4": [Circular],
                  "signup1": [Circular],
                  "signup2": [Circular],
                  "signup3": [Circular],
                  "signup4": [Circular],
                  "signup5": [Circular],
                  "subtraction": [Circular],
                  "topBannerMock1": [Circular],
                  "twitter": [Circular],
                  "umPCincoPLogo": [Circular],
                  "uncheckedBox": [Circular],
                  "unfilledHeart": [Circular],
                  "update": [Circular],
                  "visa": [Circular],
                  "whatsapp": [Circular],
                  "withoutInternet": [Circular],
                  "zoomHand": [Circular],
                },
                "default": {
                  "GuideBermuda": [Circular],
                  "GuideCalca": [Circular],
                  "GuideCamiseta": [Circular],
                  "GuideHoodie": [Circular],
                  "GuideJaqueta": [Circular],
                  "GuideMangaCurta": [Circular],
                  "GuideMangaLonga": [Circular],
                  "GuidePolo": [Circular],
                  "GuideShort": [Circular],
                  "GuideSueter": [Circular],
                  "GuideSungaCueca": [Circular],
                  "QrcodeBackground": [Circular],
                  "abandonedCartBag": [Circular],
                  "addition": [Circular],
                  "americanExpress": [Circular],
                  "arrowInstruction": [Circular],
                  "arrowRight": [Circular],
                  "bagEmpty": [Circular],
                  "bannerCatalog": [Circular],
                  "bannerLogin": [Circular],
                  "bannerOffer": [Circular],
                  "black1p5p": [Circular],
                  "cardFront": [Circular],
                  "cardImage": [Circular],
                  "cardVerso": [Circular],
                  "chat": [Circular],
                  "checkedBox": [Circular],
                  "checkedRounded": [Function],
                  "chevronLeftFill": [Circular],
                  "chevronRight": [Function],
                  "chevronRightFill": [Circular],
                  "close": [Circular],
                  "closeIcon": [Function],
                  "clothingCare1": [Circular],
                  "clothingCareBleach": [Circular],
                  "clothingCareDrying": [Circular],
                  "clothingCarePass": [Circular],
                  "clothingCareProfessionalWashing": [Circular],
                  "clothingCareWash": [Circular],
                  "config": [Circular],
                  "copy": [Circular],
                  "cupomModalBackground": [Circular],
                  "cutlery": [Circular],
                  "discountFlag": [Circular],
                  "dividerReserva": [Circular],
                  "document": [Circular],
                  "elo": [Circular],
                  "emptyProductCatalog": [Circular],
                  "expand": [Circular],
                  "facebook": [Circular],
                  "faq": [Circular],
                  "filledHeart": [Circular],
                  "foraDoAr": [Circular],
                  "green1p5p": [Circular],
                  "greenCheck": [Circular],
                  "hanger": [Function],
                  "headerLogin": [Circular],
                  "heart": [Circular],
                  "help": [Circular],
                  "homeMock1": [Circular],
                  "homeMock2": [Circular],
                  "homeMock3": [Circular],
                  "homeMock4": [Circular],
                  "imageCashback": [Circular],
                  "imageNotFound": [Circular],
                  "info": [Function],
                  "infoSuccess": [Function],
                  "instagram": [Circular],
                  "ios14": [Circular],
                  "localReserva": [Circular],
                  "logo": [Circular],
                  "logoPrime": [Function],
                  "logoQRCode": [Circular],
                  "mastercard": [Circular],
                  "newsletter": [Circular],
                  "noStoresFound": [Circular],
                  "noWishList": [Circular],
                  "personalize": [Function],
                  "pinPlace": [Circular],
                  "pinYou": [Circular],
                  "profile": [Circular],
                  "purchaseConfirmation": [Circular],
                  "raceImageBackground": [Circular],
                  "reservaIcon": [Circular],
                  "roulet": [Circular],
                  "saleOff": [Circular],
                  "search": [Circular],
                  "searchNotFound": [Circular],
                  "selectRectangle": [Circular],
                  "share": [Circular],
                  "shipping1": [Circular],
                  "shipping2": [Circular],
                  "shirt1": [Circular],
                  "shirt2": [Circular],
                  "shirt3": [Circular],
                  "shirt4": [Circular],
                  "signup1": [Circular],
                  "signup2": [Circular],
                  "signup3": [Circular],
                  "signup4": [Circular],
                  "signup5": [Circular],
                  "sizeGuideIcon": [Function],
                  "subtraction": [Circular],
                  "topBannerMock1": [Circular],
                  "trash": [Function],
                  "twitter": [Circular],
                  "umPCincoPLogo": [Circular],
                  "uncheckedBox": [Circular],
                  "unfilledHeart": [Circular],
                  "update": [Circular],
                  "visa": [Circular],
                  "whatsapp": [Circular],
                  "withoutInternet": [Circular],
                  "zoomHand": [Circular],
                },
                "svgs": {
                  "checkedRounded": [Function],
                  "chevronRight": [Function],
                  "closeIcon": [Function],
                  "hanger": [Function],
                  "info": [Function],
                  "infoSuccess": [Function],
                  "logoPrime": [Function],
                  "personalize": [Function],
                  "sizeGuideIcon": [Function],
                  "trash": [Function],
                },
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": 28,
                  "width": 160,
                },
                undefined,
              ]
            }
          />
          <Text
            fontFamily="reservaSansItalic"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Italic",
                },
                {
                  "color": "#ffffff",
                  "fontSize": 15,
                },
              ]
            }
          >
            Economize
            <Text
              fontFamily="reservaSansRegular"
              style={
                [
                  {
                    "fontFamily": "ReservaSans-Regular",
                  },
                  {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
               
              R$
              69.80
            </Text>
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignItems": "baseline",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginTop": 4,
              },
              {
                "marginLeft": -40,
              },
            ]
          }
        >
          <Text
            fontFamily="reservaSansRegular"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                },
                {
                  "color": "#9e7e2f",
                  "fontSize": 24,
                  "fontWeight": "700",
                },
              ]
            }
          >
            R$
             
            279,
          </Text>
          <Text
            fontFamily="reservaSansRegular"
            fontWeight="bold"
            style={
              [
                {
                  "fontFamily": "ReservaSans-Regular",
                  "fontWeight": "700",
                },
                {
                  "color": "#9e7e2f",
                },
              ]
            }
          >
            20
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#c4a968",
          "justifyContent": "center",
          "padding": 2,
          "width": 150,
        }
      }
    >
      <Text
        fontFamily="reservaSansRegular"
        style={
          [
            {
              "fontFamily": "ReservaSans-Regular",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        Seja
        <Text
          fontSize={18}
          style={
            {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 18,
            }
          }
          variant="descontoTag2"
        >
           
          Prime
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
